docs/federated_a_crd.md

```diff
@@ -0,0 +1,399 @@
+        spec:
+          properties:
+            template:
+              type: object
+          type: object
+        status:
+          type: object
+  version: v1alpha1
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+```
+
+The **placement type** holds information about which clusters a federated resource should be spread to:
+
+```
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: federatedbarplacements.federation.example.io
+spec:
+  group: federation.example.io
+  names:
+    kind: FederatedBarPlacement
+    plural: federatedbarplacements
+  scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            clusterNames:
+              items:
+                type: string
+              type: array
+          type: object
+        status:
+          type: object
+  version: v1alpha1
+status:federatedbarsfederatedbars
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+```
+
+The **overrides type** optionally defines how a particular resource should be varied in certain federated clusters:
+
+```
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: federatedbaroverrides.federation.example.io
+spec:
+  group: federation.example.io
+  names:
+    kind: FederatedBarOverride
+    plural: federatedbaroverrides
+  scope: Namespaced
+  validation:
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            overrides:
+              items:
+                properties:
+                  clusterName:
+                    type: string
+                  replicas:
+                    format: int32
+                    type: integer
+                type: object
+              type: array
+          type: object
+        status:
+          type: object
+  version: v1alpha1
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: null
+```
+
+The federation APIs must be created in the cluster that hosts federation:
+
+```
+$ kubectl --validate=false apply -f federatedbar_crd.yaml 
+customresourcedefinition.apiextensions.k8s.io/federatedbars.example.io created
+customresourcedefinition.apiextensions.k8s.io/federatedbarplacements.example.io created
+customresourcedefinition.apiextensions.k8s.io/federatedbaroverrides.example.io created
+
+```
+
+**Note:**<br>
+Only federated resources are propagated. Hence it's important to create a federatedbar to have an instance of the target type created in member clusters.
+
+Verify if these CRDs are working as expected by creating a test-resources of the type "federatedBar"
+
+```
+$ cat federatedbar_test.yaml 
```

Comment:
Does this work without creating the `FederatedTypeConfig`? Also this seems to be duplicate with `Create federated API resources and see them propagate`?

Reply:
It works without creating the `FederatedTypeConfig`.  I did not include that information here since it will increase the length of the document :). Yeah, thanks for catching this, let me restructure this document a little bit.